### ISSUE TITLE
Docker images missing MQTT support (--mqtt flag not compiled in)

### ISSUE THREAD

**Issue by horiacondrea** [ruvnet/RuView#872]
### Description
All published Docker images (`latest`, `v0.8.0`) are built without the `mqtt` 
feature flag, making the Home Assistant integration completely unusable via Docker.

### Error

error: unexpected argument '--mqtt' found
Usage: sensing-server --tick-ms <TICK_MS> --ui-path <UI_PATH>
--http-port <HTTP_PORT> --ws-port <WS_PORT> --bind-addr <BIND_ADDR>

### Steps to reproduce
1. Pull any published image (`latest` or `v0.8.0`)
2. Run with `--source esp32 --mqtt --mqtt-host <ip>`
3. Error above appears immediately, container keeps restarting

### Expected behaviour
The Docker image should be built with `--features mqtt` as documented 
in `docs/integrations/home-assistant.md`, which explicitly recommends 
Docker as the easiest path for non-developers.

### Environment
- Docker image: `ruvnet/wifi-densepose:latest` and `ruvnet/wifi-densepose:v0.8.0`
- Deployed via Portainer on Home Assistant

Also tried building from source using the repository Dockerfile — 
same error. The Dockerfile itself is missing `--features mqtt` 
in the cargo build command.

**Comment by ruvnet** [comment]
Taking a look

**Comment by ruvnet** [comment]
Thanks for the detailed report @horiacondrea — you're right about `v0.8.0`, and this is **already fixed in newer builds**. Here's the full picture and how to get a working image.

## What happened
- **`v0.8.0`** (tagged 2026-05-13) was built *without* `--features mqtt` — its Dockerfile build line is just `cargo build --release -p wifi-densepose-sensing-server`. So that image genuinely has no `--mqtt` flag. 👍 confirmed.
- Fixed on 2026-05-25 in `2154b6931` ("fix(docker): include HA-DISCO MQTT + cog-ha-matter; restores #794"), which added `--features mqtt` to `docker/Dockerfile.rust`.
- That fix is in **v0.9.0, v0.10.0, v0.11.0, v0.12.0** and in the current rolling `latest` (rebuilt 2026-05-30 from `main`, revision `9ad550d9` — verified to include the MQTT build + CLI flags).

So current `main` is **not** missing `--features mqtt` — that line is present today (`docker/Dockerfile.rust` line 25).

## Fix: re-pull a current image
The most likely reason `latest` still failed for you is a **stale cached image** — Docker/Portainer won't re-fetch a tag it already has locally. Force a fresh pull:

```bash
# Docker Hub
docker pull ruvnet/wifi-densepose:latest
# or GHCR
docker pull ghcr.io/ruvnet/wifi-densepose:latest

# Verify the MQTT flags are present:
docker run --rm ruvnet/wifi-densepose:latest --help | grep -i mqtt
# expect: --mqtt, --mqtt-host, --mqtt-port, --mqtt-prefix, ...
```

In **Portainer**: enable **"Always pull the image"** when recreating the container (or `docker compose pull` first), otherwise it keeps the old cached `latest`. If you prefer a pinned version, use **`ghcr.io/ruvnet/wifi-densepose:v0.12.0`** (or any tag ≥ v0.9.0) instead of `v0.8.0`.

## Then run with MQTT
```bash
docker run -d --name ruview   -e CSI_SOURCE=esp32   -p 3000:3000 -p 3001:3001 -p 5005:5005/udp   ghcr.io/ruvnet/wifi-densepose:latest   --source esp32 --mqtt --mqtt-host <your-broker-ip>
```
(env-var form also works: `-e RUVIEW_MQTT=1 -e RUVIEW_MQTT_HOST=<broker-ip>`.)

> Heads-up on a recent security change (#864): the image is now **secure-by-default** — if you don't set `RUVIEW_API_TOKEN` it auto-generates one and prints it to the logs (`docker logs ruview`). For a trusted home LAN you can keep the old open behaviour with `-e RUVIEW_ALLOW_UNAUTHENTICATED=1`.

Could you confirm the `--help | grep mqtt` output after a forced re-pull? If `--mqtt` still isn't there on a freshly-pulled `latest`, please paste the first few log lines (they include the image revision) and I'll dig in. Otherwise this should be good to close as fixed in v0.9.0+.

**Comment by horiacondrea** [comment]
Hello @ruvnet ,

After force-removing the image via `docker rmi -f` and pulling a 
completely fresh `ghcr.io/ruvnet/wifi-densepose:latest`, the --mqtt 
flag is still not present in --help output.

Full --help output shows no mqtt flags at all. The image pulled 
today (2026-05-30) is still missing MQTT support.


##Log

WiFi-DensePose sensing server

Usage: sensing-server [OPTIONS]

Options:

      --http-port <HTTP_PORT>

          HTTP port for UI and REST API [default: 8080]

      --ws-port <WS_PORT>

          WebSocket port for sensing stream [default: 8765]

      --udp-port <UDP_PORT>

          UDP port for ESP32 CSI frames [default: 5005]

      --ui-path <UI_PATH>

          Path to UI static files (repo `ui/`; from `v2/` use `../ui` or rely on auto-detect) [default: ../ui]

      --tick-ms <TICK_MS>

          Tick interval in milliseconds (default 100 ms = 10 fps for smooth pose animation) [default: 100]

      --bind-addr <BIND_ADDR>

          Bind address (default 127.0.0.1; set to 0.0.0.0 for network access) [env: SENSING_BIND_ADDR=] [default: 127.0.0.1]

      --allowed-host <HOST>

          Additional hostname (with or without `:PORT`) to permit in the `Host` header — defends loopback-bound deployments against DNS rebinding. Loopback names (`localhost`, `127.0.0.1`, `[::1]`) are always permitted implicitly. Pass multiple times to add several entries. Comma-separated values are also accepted via the `SENSING_ALLOWED_HOSTS` env var

      --disable-host-validation

          Disable Host-header validation entirely. Use only when the server sits behind a reverse proxy that already canonicalises `Host` (e.g. nginx `proxy_set_header Host`) — bare deployments stay vulnerable to DNS rebinding without it

      --source <SOURCE>

          Data source: auto, wifi, esp32, simulate [default: auto]

      --benchmark

          Run vital sign detection benchmark (1000 frames) and exit

      --load-rvf <PATH>

          Load model config from an RVF container at startup

      --save-rvf <PATH>

          Save current model state as an RVF container on shutdown

      --model <PATH>

          Load a trained .rvf model for inference

      --progressive

          Enable progressive loading (Layer A instant start)

      --export-rvf <PATH>

          Export an RVF container package and exit (no server)

      --train

          Run training mode (train a model and exit)

      --dataset <PATH>

          Path to dataset directory (MM-Fi or Wi-Pose)

      --dataset-type <TYPE>

          Dataset type: "mmfi" or "wipose" [default: mmfi]

      --epochs <EPOCHS>

          Number of training epochs [default: 100]

      --checkpoint-dir <DIR>

          Directory for training checkpoints

      --pretrain

          Run self-supervised contrastive pretraining (ADR-024)

      --pretrain-epochs <PRETRAIN_EPOCHS>

          Number of pretraining epochs (default 50) [default: 50]

      --embed

          Extract embeddings mode: load model and extract CSI embeddings

      --build-index <TYPE>

          Build fingerprint index from embeddings (env|activity|temporal|person)

      --node-positions <NODE_POSITIONS>

          Node positions for multistatic fusion (format: "x,y,z;x,y,z;...") [env: SENSING_NODE_POSITIONS=]

      --calibrate

          Start field model calibration on boot (empty room required)

      --edge-registry-url <URL>

          Override the upstream URL for the edge module registry. Set to a mirror or local file://... URL for air-gapped deployments. Empty string or --no-edge-registry disables the endpoint entirely [env: RUVIEW_EDGE_REGISTRY_URL=] [default: https://storage.googleapis.com/cognitum-apps/app-registry.json]

      --edge-registry-ttl-secs <SECS>

          Cache TTL for the edge module registry, in seconds [env: RUVIEW_EDGE_REGISTRY_TTL_SECS=] [default: 3600]

      --no-edge-registry

          Disable the edge module registry endpoint entirely. Returns 404 on `GET /api/v1/edge/registry`. Use for air-gapped deployments [env: RUVIEW_NO_EDGE_REGISTRY=]


  -h, --help

**Comment by ruvnet** [comment]
@horiacondrea you were right — and thanks for the careful re-test. A fresh `latest` genuinely had **no `--mqtt` flag**, and the earlier "fixed in v0.9.0+" was a misdiagnosis on our side.

## The actual root cause (not the Docker build flag)
The `--mqtt*` flags were defined only in a `cli::Args` struct that **nothing referenced**, while the binary parses a *separate* `main::Args` that had no mqtt fields — and `main.rs` never started the `mqtt::` publisher. So `--mqtt` could never appear in `--help`, and **no Docker rebuild (`--features mqtt` or otherwise) could fix it** — it was a code-wiring gap, not a build-flag problem. That's why re-pulling didn't help.

## Fix (now merged to `main`)
PR #874 (merged) extracts the flags into a shared `cli::MqttArgs`, flattens them into the binary's real parser, spawns the publisher on `--mqtt`, and bridges the sensing broadcast into the typed `VitalsSnapshot` stream. Verified end-to-end against a live `mosquitto`: 20 Home-Assistant auto-discovery entities + live state (presence / person-count / breathing / heart-rate / RSSI).

## What you need to do
The rolling `latest` image rebuilds from `main`, so the next publish will include the wired-in flags. After it builds, please:
```bash
docker pull ghcr.io/ruvnet/wifi-densepose:latest   # force a fresh pull
docker run --rm ghcr.io/ruvnet/wifi-densepose:latest --help | grep -i mqtt
# expect: --mqtt, --mqtt-host, --mqtt-port, --mqtt-prefix, ...
```
If `--mqtt` still isn't there on a freshly-pulled image *after* the next build, paste the first log lines (they include the image revision) and I'll dig in. Sorry for the earlier runaround.
